integrated_channels/moodle/client.py

```diff
@@ -36,9 +36,10 @@ def inner(self, *args, **kwargs):
             # This cannot be converted to JSON thus the above fails miserably.
             # The above can fail with different errors depending on the format of the returned page.
             # Moodle of course does not tell us what is wrong in any part of this HTML.
-            raise ClientError('Moodle API task "{method}" failed due to unknown error.'.format(
-                method=method.__name__), response.status_code
-            ) from error
+            log_msg = (f'Moodle API task "{method.__name__}" '
+                       f'for enterprise_customer_uuid "{self.enterprise_configuration.enterprise_customer.uuid}" '
```

Comment:
let's hope this logging helps. It does contain the enterprise customer id but will we know which catalog, course etc was being processed? I guess this wrapper currently cannot access those so.. fwiw this logging on its own is useful. I just wonder if we also need more logging around other places (we can do this later too) to pinpoint course, catalog, user etc. The `generate_formatted_log` is an attempt to standardize that .. but it cannot always be used without sourcing all the required inputs

Reply:
also if this is an example of this 
```
 response = self._get_courses(serialized_data['courses[0][shortname]'])
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/integrated_channels/moodle/client.py", line 39, in inner
    raise ClientError('Moodle API task "{method}" failed due to unknown error.'.format(
integrated_channels.exceptions.ClientError: Moodle API task "_get_courses" failed due to unknown error.
```

then perhaps printing out what we can of the response may also be helpful?

Reply:
`then perhaps printing out what we can of the response or response code from moodle`

> Moodle of course does not tell us what is wrong in any part of this HTML.

Sounds like Talia found out that the response body really doesn't give us much. The code is getting logged here, and I'm not entirely sure if we're able to get more (useful) information out of it.